### PR TITLE
Improve error message for no GraphQL Port

### PR DESCRIPTION
**Pull request by zionts**
This also allows ports with value '0' and disallows non-numeric ports

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -136,11 +136,15 @@ export class Engine extends EventEmitter {
...
         } else {
-            const port = process.env.PORT;
-            if (port) {
+            const port : any = process.env.PORT;
+            if (!isNaN(port) && isFinite(port)) {
                 this.graphqlPort = parseInt(port, 10);
             } else {
-                throw new Error('process.env.PORT is not set!');
+                throw new Error(`Neither 'graphqlPort' nor process.env.PORT is set. ` +
+                    `In order for Apollo Engine to act as a proxy for your GraphQL server,` +
```
Comment (thepwagner) [inline src/index.ts]: Only the first line has a trailing space, so this will join `server,it`?

File: src/index.ts [hunk 2]
```diff
@@ -136,11 +136,15 @@ export class Engine extends EventEmitter {
         if (config.graphqlPort) {
             this.graphqlPort = config.graphqlPort;
         } else {
-            const port = process.env.PORT;
-            if (port) {
+            const port : any = process.env.PORT;
+            if (!isNaN(port) && isFinite(port)) {
```
Comment (thepwagner) [inline src/index.ts]: Since `isFinite(NaN)===false`, can this be simplified?

File: src/index.ts [hunk 3]
```diff
@@ -136,11 +136,15 @@ export class Engine extends EventEmitter {
...
                 this.graphqlPort = parseInt(port, 10);
             } else {
-                throw new Error('process.env.PORT is not set!');
+                throw new Error(`Neither 'graphqlPort' nor process.env.PORT is set. ` +
+                    `In order for Apollo Engine to act as a proxy for your GraphQL server,` +
+                    `it needs to know which port your GraphQL server is (this is the port number` +
```
Comment (thepwagner) [inline src/index.ts]: Wording nit: `... server is listening on` reads better to me than just `... server is`

File: src/index.ts (hunk 2 above)
Comment (zionts) [reply src/index.ts]: isFinite(undefined) will throw an error, whereas isNaN will not

Comment (zionts) [reply src/index.ts]: Actually, I think that's not true^ I think you're right that we can just use isFinite(port)

File: src/index.ts (hunk 1 above)
Comment (zionts) [reply src/index.ts]: 👍

File: src/index.ts (hunk 3 above)
Comment (zionts) [reply src/index.ts]: Whoops. Thought I wrote that. Fixed 👂

**Comment by zionts** [comment]
Merging without tests for now

**Pull request merged**